pulsar-client-cpp/tests/ConsumerConfigurationTest.cc

```diff
@@ -28,11 +28,19 @@ DECLARE_LOG_OBJECT()
 
 using namespace pulsar;
 
+class DummyEventListener : public ConsumerEventListener {
+   public:
+    virtual void becomeActive(Consumer consumer, int partitionId) override {}
```

Comment:
We should add a test that verifies that the active/inactive tests are being invoked. We can follow the same line of test that we're doing for java

Reply:
I have added unit tests in `ConsumerTest`. And I renamed "become" to "became" to make it consistent with java client. The call to `registerConsumer` is moved from `handleCreateConsumer` to `connectionOpened`, so that we can handle commands such as `ACTIVE_CONSUMER_CHANGE` properly.

@merlimat Please take a look.